### PR TITLE
Alphabetize editor plugins in EditorNode and move 2D plugins to their own section

### PR DESCRIPTION
**Pull request by aaronfranke**
Before this PR, the list of editor plugins added using `add_editor_plugin` seemed to be mostly in a random order. This PR alphabetizes them so that they have a consistent and predictable order.

I also split the 2D ones to be by themselves. This section is specifically for editor plugins that depend on Node2D, so plugins that work with Control nodes don't go here. This is useful for some future work, and is why I decided to make this PR. Later we can do the same with plugins that depend on Node3D, but I don't have a comprehensive list of those off hand (and it's not as simple as checking which have "3D" in the name).

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -7073,63 +7073,65 @@ EditorNode::EditorNode() {
 
 	// More visually meaningful to have this later.
 	raise_bottom_panel_item(AnimationPlayerEditor::get_singleton());
-	add_editor_plugin(memnew(ReplicationEditorPlugin));
 
 	add_editor_plugin(VersionControlEditorPlugin::get_singleton());
```
Comment (akien-mga) [inline editor/editor_node.cpp:7077]: I'd suggest changing that one too to follow the same pattern as others.
It implements the singleton pattern this way, which is not a bad idea, but it's inconsistent with the common usage in Godot:
```cpp
VersionControlEditorPlugin *VersionControlEditorPlugin::get_singleton() {
        return singleton ? singleton : memnew(VersionControlEditorPlugin);
}
```

Make that just `return singleton;` and you can `memnew` it here instead, like other plugins.

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -7073,63 +7073,65 @@ EditorNode::EditorNode() {
...
-	add_editor_plugin(memnew(ShaderEditorPlugin));
-	add_editor_plugin(memnew(ShaderFileEditorPlugin));
 
+	add_editor_plugin(memnew(AnimationTreeEditorPlugin));
```
Comment (akien-mga) [inline editor/editor_node.cpp:7079]: I'd add a comment to introduce the editor plugins block, make clear that it should be ordered alphabetically in the appropriate category.

My experience shows that a significant proportion of contributors don't infer "alphabetical ordering" when they see a clearly alphabetically ordered list.

File: editor/editor_node.cpp (hunk 1 above)
Comment (aaronfranke) [reply editor/editor_node.cpp:7077]: I just tried this, but it crashes when the editor is started.

File: editor/editor_node.cpp (hunk 2 above)
Comment (aaronfranke) [reply editor/editor_node.cpp:7080]: Done, added a comment.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**